lib/simulator/SimulatorLightController.cpp

```diff
@@ -10,92 +8,58 @@
 #include <utility>
 #include <vector>
 
-SimulatorLightController::SimulatorLightController(Ogre::SceneManager *scnMgr)
-    : scnMgr(scnMgr) {
-  for (int pole_index = 0; pole_index < 6; pole_index++) {
-    auto pole = Pole();
-    const unsigned long grid_num_rows = pole.get_grid_lights().size();
-    const unsigned long grid_num_cols = pole.get_grid_lights().front().size();
-
-    std::vector<std::vector<Ogre::Light *>> light_pole;
-    for (unsigned long row = 0; row < grid_num_rows; row++) {
-      std::vector<Ogre::Light *> light_row;
-      for (unsigned long col = 0; col < grid_num_cols; col++) {
-        Ogre::Vector3 unscaled = Ogre::Vector3(
-            kCircleRadiusInches * cos(M_PI / 3.0 * pole_index) +
-                col * kPixelPitchInches * sin(M_PI / 3.0 * pole_index),
-            kPoleHeightInches - row * kPixelPitchInches,
-            kCircleRadiusInches * sin(M_PI / 3.0 * pole_index) +
-                col * kPixelPitchInches * cos(M_PI / 3.0 * pole_index));
-
-        Ogre::Vector3 scaled =
-            Ogre::Vector3(SimulatorLightController::inchesToCoords(unscaled.x),
-                          SimulatorLightController::inchesToCoords(unscaled.y),
-                          SimulatorLightController::inchesToCoords(unscaled.z));
-        light_row.push_back(createLight(scaled));
-      }
-      light_pole.push_back(std::move(light_row));
-    }
-    poles.push_back(std::move(pole));
-    lights.push_back(std::move(light_pole));
-  }
+SimulatorLightController::SimulatorLightController() {
+  led_pixels_ = kLedPixels;
```

Comment:
I would prefer an in-class initializer if a variable should always be initialized to some constant value.

Reply:
Changed the API to be a method with this class overrides.